cpp/src/arrow/util/compression.h

```diff
@@ -107,6 +107,46 @@ class ARROW_EXPORT Decompressor {
   // XXX add methods for buffer size heuristics?
 };
 
+/// \brief Compression codec options
+class ARROW_EXPORT CodecOptions {
+ public:
+  CodecOptions(int compression_level = kUseDefaultCompressionLevel) {
+    compression_level_ = compression_level;
+  }
+  virtual ~CodecOptions() = default;
+
+  int compression_level_;
+};
+
+// ----------------------------------------------------------------------
+// gzip codec options implementation
+
+struct GZipFormat {
+  enum type {
+    ZLIB,
+    DEFLATE,
+    GZIP,
+  };
+};
+
+constexpr int kGZipDefaultWindowBits = 15;
+
+class ARROW_EXPORT GZipCodecOptions : public CodecOptions {
+ public:
+  GZipFormat::type gzip_format = GZipFormat::GZIP;
+  int window_bits = kGZipDefaultWindowBits;
+};
+
+// ----------------------------------------------------------------------
+// brotli codec options implementation
+
+constexpr int kBrotliDefaultWindowBits = 22;
```

Comment:
Can it use `BROTLI_DEFAULT_WINDOW`?

Reply:
yes, but then we need to include the `brotli` header here as well?

Reply:
yes, I think that's just how it is implemented now in `GZip/BrotliCodecOptions`?

Reply:
Sure, I notice that this patch hard code the encoder's internal into `compression.h`, so I'm afraid if they're changed in the 3rd libraries, we'll get confused here.

Reply:
yeah it seems better to use their `BROTLI_DEFAULT_WINDOW`

Reply:
Same with my previous comment. We can use `BROTLI_DEFAULT_WINDOW` in the `.cc` file